### PR TITLE
wxGUI: Fixed bare 'except' for psmap/

### PR DESCRIPTION
**Pull request by arohanajit**
Replace bare except with specific font creation error handling
- Adds `except (ValueError, wx.PyAssertionError)` in `makePSFont()` to handle invalid font parameters and system font failures

Improve text measurement error handling 
- Adds `except (wx.PyAssertionError, ValueError)` in `getTextExtent()` for device context and text measurement failures

### CODE REVIEW

**Comment by echoix** [comment]
Can the title better represent what is changed here? E722 isn't fixed for all wxGUI, only one file is fixed here and other wxGUI remains

File: gui/wxpython/psmap/frame.py [hunk 1]
```diff
@@ -1049,7 +1049,7 @@ def makePSFont(self, textDict):
             fn = wx.Font(
                 pointSize=fontsize, family=family, style=style, weight=weight, face=face
```
Comment (petrasovaa) [inline gui/wxpython/psmap/frame.py:1050]: Looks like the parameters need to be updated:
```suggestion
                pointSize=fontsize, family=family, style=style, weight=weight, faceName=face
```
see https://docs.wxpython.org/wx.Font.html#wx.Font.__init__

File: gui/wxpython/psmap/frame.py [hunk 2]
```diff
@@ -1071,7 +1071,7 @@ def getTextExtent(self, textDict):
             dc.SetFont(fn)
             w, h, lh = dc.GetFullMultiLineTextExtent(textDict["text"])
             return (w, h)
-        except:
+        except (wx.PyAssertionError, ValueError):
```
Comment (petrasovaa) [inline gui/wxpython/psmap/frame.py:1074]: KeyError?

File: gui/wxpython/psmap/frame.py [hunk 3]
```diff
@@ -1049,7 +1049,7 @@ def makePSFont(self, textDict):
             fn = wx.Font(
                 pointSize=fontsize, family=family, style=style, weight=weight, face=face
             )
-        except:
+        except (ValueError, wx.PyAssertionError):
```
[the remark this answers is not in the document]
Comment (arohanajit) [reply gui/wxpython/psmap/frame.py:1052]: `wx.PyAssertionError` to handle any errors on the lib side
`ValueError` if any of the passed parameters are invalid

[the remark this answers is not in the document]
Comment (petrasovaa) [reply gui/wxpython/psmap/frame.py:1056]: I doubt these would be raised. I couldn't get it throw any exception except TypeError, which would be caused by programmer's error. So I would just remove the try except block, we don't use it anywhere else with wx.Font

**Review by petrasovaa — APPROVED** [review]

**Pull request merged**